tools/update-prod-static

```diff
@@ -28,7 +29,15 @@ subprocess.check_call(["mkdir", "-p", "var/log"])
 fp = open('var/log/update-prod-static.log', 'w')
 
 # Install node packages
-subprocess.check_call(['npm', 'install', '--production'], stdout=fp, stderr=fp);
+exists, success_stamp = check_node_cache()
+if not exists:
+    print("Deleting cached version")
+    subprocess.check_call(["rm", "-rf", "node_modules"], stdout=fp, stderr=fp)
+    print("Installing node modules")
+    subprocess.check_call(['npm', 'install', '--production'], stdout=fp, stderr=fp)
+    subprocess.check_call(["mkdir", "-p", success_stamp], stdout=fp, stderr=fp)
+else:
+    print("Using cached version of node_modules")
```

Comment:
The way that we do deployments in production, each deployment has its own Zulip checkout.  So I think we should do a flow that looks more like how virtualenvs work with a unified system used in both dev and prod:
* We make something like /srv/zulip-npm-cache (or /home/travis/npm-cache in Travis CI) where we actually do the `npm` builds in a directory named after the `package.json` sha1 + whether `--production` was passed as an argument.
* This code checks if the right node_modules directory is in the cache, and if it is, just moves the node_modules symlink to point to it.  And if it isn't, it does the `npm install`
* We'd add some garbage collection code similar to `tools/clean-venv-cache` (ideally reusing some of the code) so the caches in Travis CI don't grow without bound.

Does that make sense?  I'm not sure this is perfect but it seems like it would solve the problem in a robust way.